client/src/views/home/HomePage.vue

```diff
@@ -360,10 +361,17 @@ async function handleQuery(): Promise<void> {
     }
   } else if (query.bmsOrganizationId) {
     const availableDevices = await listAvailableDevices();
-    const device = availableDevices.find((d) => {
-      const serverType = getServerTypeFromAddress(d.serverUrl);
-      return serverType === ServerType.Saas && d.organizationId === query.bmsOrganizationId;
-    });
+    let device = undefined;
+    for (const d of availableDevices) {
+      const result = await parseParsecAddr(d.serverAddr);
+      if (!result.ok) {
+        throw Error(`Invalid \`serverAddr\` field for AvailableDevice: ${d}`);
```

Comment:
Use `continue` instead of throwing an error. Any call to libparsec that fails is automatically logged so we're covered, might as well not interrupt the execution.

Reply:
Both exceptions and a function from libparsec returning an error (`.ok` is false) are logged. This can indeed lead to false positives but it's fine. You can take a look at https://github.com/Scille/parsec-cloud/blob/dee9e1c6b021df4fde4d2ab2148c6b315b457cd9/client/src/plugins/libparsec/index.ts#L51

For this specific error, I think continuing "silently" (it's still logged) is our best bet. We don't really have anything to show the user, we don't want to display an error, we don't want to interrupt the function.